Darwin/Examples/ToDoSync/ToDoSync/TodoTableViewController.swift

```diff
@@ -159,63 +154,67 @@ class TodoTableViewController:
     @objc func removeAll(_ sender: Any) {
         itemsCollection.deleteMany(["owner_id": userId!,
                                     "checked": true]) { result in
-            switch result {
-            case .failure(let error):
-                print(error.localizedDescription)
-            case .success(_):
-                listsCollection.sync.updateOne(
-                    filter: ["_id": self.userId ?? BSONNull()],
-                    update: ["$set": ["todos": self.todoItems.compactMap({ !$0.checked ? $0.id : nil })] as Document], options: nil) { _ in
-                    DispatchQueue.main.sync {
-                        self.checkBoxAll.on = false
-                    }
-                }
-            }
+                                        switch result {
+                                        case .failure(let error):
+                                            print(error.localizedDescription)
+                                        case .success(_):
+                                            listsCollection.sync.updateOne(
+                                                filter: ["_id": self.userId ?? BSONNull()],
+                                                update: ["$set": ["todos": self.todoItems.compactMap({ !$0.checked ? $0.id : nil })] as Document], options: nil) { _ in
+                                                    DispatchQueue.main.sync {
+                                                        self.checkBoxAll.on = false
+                                                    }
+                                            }
+                                        }
         }
     }
 
-    private func loggedIn() {
-        if listsCollection.sync.syncedIds.isEmpty {
-            listsCollection.sync.insertOne(document: TodoList(id: userId!)) { _ in }
+    private func loggedIn() { async {
```

Comment:
[broader discussion, does not need be addressed right now] if this hacky workaround stuff has to be done for usage of our API to be clean, we should perhaps rethink the API before GA. Maybe it makes sense to offer blocking versions of certain things 🤷‍♂️

Reply:
We should probably adopt either PromiseKit or Async/Await.